newswires/client/src/SideNav.tsx

```diff
@@ -1,47 +1,93 @@
 import {
+	EuiBadge,
+	EuiBadgeGroup,
 	EuiCollapsibleNav,
 	EuiCollapsibleNavGroup,
 	EuiHeaderSectionItemButton,
 	EuiIcon,
 	EuiListGroup,
+	EuiListGroupItem,
+	EuiText,
 } from '@elastic/eui';
-import { useMemo, useState } from 'react';
+import { css } from '@emotion/react';
+import { useCallback, useMemo, useState } from 'react';
+import { brandColours } from './sharedStyles';
 import type { Query } from './sharedTypes';
 import { useSearch } from './useSearch';
 
-interface MenuItem {
-	label: string;
-	query: Query;
+function decideLabelForQueryBadge(query: Query): string {
+	const { supplier, q, subject } = query;
+	const supplierLabel = supplier.join(', ');
+	const qLabel = q.length > 0 ? `"${q}"` : '';
+	const subjectLabel = subject.join(', ');
+	const labels = [supplierLabel, qLabel, subjectLabel];
+	return labels.filter((label) => label.length > 0).join(' ');
 }
 
+const recognisedSuppliers = ['REUTERS', 'AP', 'AAP', 'PA'];
+
 export const SideNav = () => {
 	const [navIsOpen, setNavIsOpen] = useState<boolean>(false);
 
-	const { state } = useSearch();
+	const { state, config, handleEnterQuery } = useSearch();
 
 	const searchHistory = state.successfulQueryHistory;
+	const activeSuppliers = config.query.supplier;
 
-	const searchHistoryItems: MenuItem[] = useMemo(
+	const searchHistoryItems = useMemo(
 		() =>
-			searchHistory.map(({ query, resultsCount }) => ({
-				label: `${query.q} (${resultsCount})`,
+			searchHistory.slice(1).map(({ query, resultsCount }) => ({
+				label: decideLabelForQueryBadge(query),
 				query,
+				resultsCount,
 			})),
 		[searchHistory],
 	);
 
-	const agencies: MenuItem[] = [
-		{ label: 'All', query: { q: '' } },
-		{ label: 'Reuters', query: { q: '', supplier: ['REUTERS'] } },
-		{ label: 'AP', query: { q: '', supplier: ['AP'] } },
-		{ label: 'PA', query: { q: '', supplier: ['PA'] } },
-		{ label: 'AAP', query: { q: '', supplier: ['AAP'] } },
-	];
+	const toggleSupplier = useCallback(
+		(supplier: string) => {
+			// If 'activeSuppliers' is empty, that means that *all* suppliers are active.
+			if (activeSuppliers.length === 0) {
+				handleEnterQuery({
+					...config.query,
+					supplier: recognisedSuppliers.filter((s) => s !== supplier),
+				});
```

Comment:
hmm I think this behaves the opposite to how I would expect - I think if all suppliers are active and I click on AP, I'd expect to filter to just AP, not filter to everything except AP (maybe adding a shift+click or alt+click handler to exclude AP)

Reply:
Alt would be excluding (as per recent Grid precedent, itself modelled on decades-long Adobe one)